### ISSUE TITLE
[Bug]: `withServices`: Huawei Nova 8 SE cannot scan using `withServices`, but version 1.20.7 works.

### ISSUE THREAD

**Issue by stengfei** [boskokg/flutter_blue_plus#710]
### Requirements

- [X] I've looked at the README 'Common Problems' section

### Have you checked this problem on the example app?

No

### FlutterBluePlus Version

1.29.8

### Flutter Version

3.1.0

### What OS?

Android

### OS Version

NOVA 8 SE HarmonyOS

### Bluetooth Module

unknow

### What is your problem?

Some mobile phones, such as Huawei Nova 8 SE, cannot scan any devices using service filtering method (startScan using withServices parameter). Version 1.20.7 does not have this problem.

### Logs

```shell
```

**Comment by stengfei** [comment]
The problem started in version 1.27.0, and version 1.26.6 was the last version without this problem.

**Comment by chipweinberger** [comment]
which version did it start?

1.26.6? or 1.27.0?

**Comment by chipweinberger** [comment]
try setting `continuousUpdates:true`

**Comment by stengfei** [comment]
The problem started in version 1.27.0. Setting continuousUpdates: true can solve the problem.

**Comment by stengfei** [comment]
> Hmm thats not good because `continousUpdates` takes more power and will slow down your UI. `withServices` is supposed to work without it.
> 
> Can you turn `continousUpdates` off, turn on verbose logs, and add print statements to this code?
> 
> ```
>                 // filter duplicates
>                 if (((boolean) mScanFilters.get("continuous_updates")) == false) {
>                     boolean isDuplicate = mAdvSeen.containsKey(remoteId) && mAdvSeen.get(remoteId).equals(advHex);
>                     mAdvSeen.put(remoteId, advHex); // remember
>                     // print the value of advHex
>                     if (isDuplicate) {
>                         // print here
>                         return;
>                     }
>                 }
> ```


I'm not quite sure if the position of this code I added is correct, as shown in the picture below
![image](https://github.com/boskokg/flutter_blue_plus/assets/22679878/f854f4c1-53ad-45af-874c-6dbd13d3ff2a)

**Comment by chipweinberger** [comment]
```
                // filter duplicates
                if (((boolean) mScanFilters.get("continuous_updates")) == false) {
                    boolean isDuplicate = mAdvSeen.containsKey(remoteId) && mAdvSeen.get(remoteId).equals(advHex);
                    mAdvSeen.put(remoteId, advHex); // remember
                    log(logLevel.DEBUG, "advHex:" + advHex);
                    if (isDuplicate) {
                        log(logLevel.DEBUG, "isDuplicate");
                        return;
                    }
                }
```

**Comment by stengfei** [comment]
> I'd add some logs like this.
> 
> And then maybe you can debug why you need to set `continuousUpdates` true to get any scan results. you're supposed to get results even when `continuousUpdates` is false.
> 
> ```
>                 // filter duplicates
>                 if (((boolean) mScanFilters.get("continuous_updates")) == false) {
>                     boolean isDuplicate = mAdvSeen.containsKey(remoteId) && mAdvSeen.get(remoteId).equals(advHex);
>                     mAdvSeen.put(remoteId, advHex); // remember
>                     log(logLevel.DEBUG, "advHex:" + advHex);
>                     if (isDuplicate) {
>                         log(logLevel.DEBUG, "isDuplicate");
>                         return;
>                     }
>                 }
> ```

Thanks, I'll try it next monday

**Comment by chipweinberger** [comment]
these logs are better

```
                log(logLevel.DEBUG, remoteId + " advHex:" + advHex); 

                // filter duplicates
                if (((boolean) mScanFilters.get("continuous_updates")) == false) {
                    boolean isDuplicate = mAdvSeen.containsKey(remoteId) && mAdvSeen.get(remoteId).equals(advHex);
                    mAdvSeen.put(remoteId, advHex); // remember
                    if (isDuplicate) {
                        log(logLevel.DEBUG, "filtered duplicate. " + remoteId);
                        return;
                    }
                }

                    // filter keywords
                    String name = scanRecord != null ? scanRecord.getDeviceName() : "";
                    List<String> keywords = (List<String>) mScanFilters.get("with_keywords");
                    if (filterKeywords(keywords, name) == false) {
                        log(logLevel.DEBUG, "filtered keyword " + remoteId);
                        return;
                    }

                    // filter divisor
                    if (((boolean) mScanFilters.get("continuous_updates")) != false) {
                        int count = scanCountIncrement(remoteId);   
                        int divisor = (int) mScanFilters.get("continuous_divisor");
                        if ((count % divisor) != 0) {
                            log(logLevel.DEBUG, "filtered divisor " + remoteId);
                            return;
                        }
                    }
```

**Comment by chipweinberger** [comment]
fixed 1.29.12
